checks.d/dns_check.py

```diff
@@ -20,66 +20,96 @@
     from dns.rdtypes.ANY import *  # noqa
     from dns.rdtypes.IN import *  # noqa
 
+class BadConfException(Exception):
+    pass
 
-class DNSCheck(AgentCheck):
+class DNSCheck(NetworkCheck):
     SERVICE_CHECK_NAME = 'dns.can_resolve'
     DEFAULT_TIMEOUT = 5
 
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
-        self.default_timeout = init_config.get('default_timeout', self.DEFAULT_TIMEOUT)
+    def __init__(self, name, init_config, agentConfig, instances):
```

Comment:
I did this to set the `default_timeout` when configured in `init_config`. `network.py` only passes `instance` to `_check()`, not `init_config`. 
https://github.com/DataDog/dd-agent/blob/master/checks/network_checks.py#L138

Is there a better way to set `default_timeout` when configured in the yaml file?